src/ui/Analytics/LiveAnalyticsClient.ts

```diff
@@ -269,10 +269,26 @@ export class LiveAnalyticsClient implements IAnalyticsClient {
     return this.resolveActiveTabFromElement(element) || 'default';
   }
 
-  private buildMetaObject<TMeta>(meta: TMeta): IChangeableAnalyticsMetaObject {
-    var build: IChangeableAnalyticsMetaObject = _.extend({}, meta);
-    build['JSUIVersion'] = version.lib + ';' + version.product;
-    return build;
+  private buildMetaObject<TMeta>(meta: TMeta, result?: IQueryResult): IChangeableAnalyticsMetaObject {
+    let modifiedMeta: IChangeableAnalyticsMetaObject = _.extend({}, meta);
+    modifiedMeta['JSUIVersion'] = version.lib + ';' + version.product;
+
+    if (result) {
+      let fieldValue;
+      let fieldUsed;
+      let uniqueId = result.raw['uniqueid'];
+      if (uniqueId) {
+        fieldUsed = 'uniqueid';
+        fieldValue = uniqueId;
+      } else {
+        fieldUsed = 'urihash';
+        fieldValue = result.raw['urihash'];
+      }
+      modifiedMeta['contentIdKey'] = fieldUsed;
```

Comment:
Erhm, I asked Seb Paquet, he told me to use contentIdKey and contentIdValue (lower case d).

I also checked the coveo.analytics lib (that is used for the recommendation feature), and it is also a lower case d.

So I guess you should instead change the casing in the backend, no ?